### ISSUE TITLE
Recognition of small group does not terminate

### ISSUE THREAD

**Issue by hulpke** [gap-packages/recog#6]
mats:= [[ [ 0, 0, 1 ], [ 1, 0, 0 ], [ 0, 1, 0 ] ],
  [ [ 1, 22, 28 ], [ 0, 14, 10 ], [ 0, 13, 16 ] ]]*Z(31)^0;
g:=Group(mats);

Then g is irreducible and has order 1080 (3.A6 which is for p=31 a non geometric maximal of SL_3).

RecognizeGroup seems to be stuck in `RecognizeGeneric'.

Alexander

**Comment by fingolfin** [comment]
The recognition now works due to the change I made in de53b9090c13536. But I am not sure whether this fixes the actual bug, or merely coveres it up.

What I did was to remove a "hint" for L2(31). It is striking that this was the only "hint" of this form in the system for a non-sporadic group. Its origin is hidden in the mists of time (i.e. the repository history does not tell anything useful, as it was added along with the surrounding code). 

Some theories about this bug:
* the hint was simply "wrong", and thus lead to this fix. Thus, removing it constitutes a proper fix, and this issue can be closed.
* the hint was correct, but some other bug causes the system to use it inappropriately, i.e. when it should not have used the hint in the first place
* the hint was correct, and recog was right in using it. However, the code which uses hints needs to be aware that hints can be wrong, and thus should abort trying to use the hint after a certain time. I.e. the infinite loop (see commit 427a77296) should abort after, say 500 iterations, if no suitable element was found, and should somehow signal the caller that the hint was no good, and something else should be tried.
* ... perhaps yet another explanation is correct... ?
